cron/monthly/verify-plans.js

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env node
+import '../../server/env';
+
+/**
+ * Makes sure all collectives.plan are up-to-date, downgrading and cancelling plans if needed.
+ */
+
+// Only run on the first of the month
+const today = new Date();
+if (process.env.NODE_ENV === 'production' && today.getDate() !== 1) {
+  console.log('NODE_ENV is production and today is not the first of month, script aborted!');
+  process.exit();
+}
+
+process.env.PORT = 3066;
+
+import { Op } from 'sequelize';
+import { compact, findKey, get, groupBy, map, values, pipe } from 'lodash/fp';
+import debugLib from 'debug';
+
+import models from '../../server/models';
+import { promiseSeq } from '../../server/lib/utils';
+import plans, { PLANS_COLLECTIVE_SLUG } from '../../server/constants/plans';
+import orderStatus from '../../server/constants/order_status';
+import emailLib from '../../server/lib/email';
+
+const debug = debugLib('verify-plans');
+
+const d = new Date();
+d.setMonth(d.getMonth() - 1);
+
+const START_DATE = new Date(d.getFullYear(), d.getMonth(), 1);
+const END_DATE = new Date(d.getFullYear(), d.getMonth() + 1, 1);
+const REPORT_EMAIL = 'ops@opencollective.com';
+const BATCH_SIZE = 10;
+const EXISTING_PLANS_SLUGS = pipe(values, map(get('slug')), compact)(plans);
+
+const LEVELS = {
+  LEGACY: 'LEGACY',
+  DOWNGRADE: 'DOWNGRADE',
+  CANCEL: 'CANCEL',
+  EXCEPTION: 'EXCEPTION',
+};
+
+export async function run(options = {}) {
+  const collectives = await models.Collective.findAll({ where: { plan: { [Op.ne]: null } } });
+  debug(`There is/are ${collectives.length} subscribed to our plans...`);
+
+  const info = [];
+
+  await promiseSeq(
+    collectives,
+    async c => {
+      debug(`Processing collective #${c.id}...`);
+      // Custom or legacy plans, we're ignoring this because this was manually set.
+      if (!plans[c.plan].slug) {
+        return info.push({
+          level: LEVELS.LEGACY,
+          message: `${c.slug} is using legacy plan, ignoring.`,
+        });
+      }
+
+      const [lastOrder] = await models.Order.findAll({
+        include: [
+          { model: models.Collective, as: 'collective', where: { slug: PLANS_COLLECTIVE_SLUG } },
+          { model: models.Collective, as: 'fromCollective', where: { id: c.id } },
+          { model: models.Subscription, as: 'Subscription' },
+          { model: models.Tier, as: 'Tier', where: { slug: { [Op.in]: EXISTING_PLANS_SLUGS } } },
+        ],
+        limit: 1,
+        order: [['updatedAt', 'DESC']],
+      });
+
+      const lastOrderPlan = findKey({ slug: lastOrder.Tier.slug }, plans);
```

Comment:
Is it because it's the fp version of findKey? Doesn't match the documentation. https://lodash.com/docs/4.17.15#findKey

```findKey(users, { 'age': 1, 'active': true });```

Reply:
Yes, in the functional programming version you first create functions and then invoke with the data.
I'll replace this with pure JS like suggested above.